### PR TITLE
Bug 1560633 - add support for xml2js

### PR DESCRIPTION
**Pull request by djmitche**
This adds xml2js, after adding a bunch of tests to ensure compatibility.

Among other things, this allows us to use this library on node 12, where libxmljs doesn't have precompiled binaries (and, in fact, where libxmljs doesn't even build).

[libxmljs](https://github.com/libxmljs/libxmljs) and [pixl-xml](https://github.com/jhuckaby/pixl-xml) are both unmaintained.  I'm tempted to say we should just drop support for them.  My only concern is, since speed is not a priority for xml2js, are we in so doing sacrificing the "fast" in "fast-azure-storage"?  In that case, would we be better off rewriting azure-{entities,blob-storage}to use the upstream MS libs?

@jonasfj perhaps you have some thoughts here?

### CODE REVIEW

**Comment by imbstack** [comment]
Oh, I see they were hidden behind a "large diff" thing. /me continues looking

File: lib/xml-parser/xml2js-parser.js [hunk 1]
```diff
@@ -0,0 +1,428 @@
+const xml2js = require('xml2js');
+
+// xml2js uses a callback calling pattern for a sync operation.  Ok..  happily,
+// the `async: false` forces it to call back before returning, so we can just
+// wrap it to make it sync.
```
[the remark this answers is not in the document]
Comment (djmitche) [reply lib/xml-parser/xml2js-parser.js]: It does default that way but the [docs](https://github.com/Leonidas-from-XIV/node-xml2js#options) say they may change the default.  Which is just weird: it's a totally sync operation, so why?  I think it has to do with a confusion between the callback style of the SAX DOM parser they use and what "async" means.  There's no other part of the lib that is actually async.

Anyway, if xml2js messes this up in some future upgrade, we'll just go looking for another XML library.  I think the comment is accurate as it stands.

**Comment by djmitche** [comment]
I found a few more bugs in the new parser :(

How's that look?

**Comment by djmitche** [comment]
Hm, Travis seems to not be running..

**Comment by djmitche** [comment]
Oh, right, it was failing consistently.

**Pull request merged**